modin/pandas/test/dataframe/test_binary.py

```diff
@@ -167,28 +167,68 @@ def test_multi_level_comparison(data, op):
         getattr(modin_df_multi_level, op)(modin_df_multi_level, axis=0, level=1)
 
 
-def test_equals():
-    frame_data = {"col1": [2.9, 3, 3, 3], "col2": [2, 3, 4, 1]}
-    modin_df1 = pd.DataFrame(frame_data)
-    modin_df2 = pd.DataFrame(frame_data)
-
-    assert modin_df1.equals(modin_df2)
-
-    df_equals(modin_df1, modin_df2)
-    df_equals(modin_df1, pd.DataFrame(modin_df1))
-
-    frame_data = {"col1": [2.9, 3, 3, 3], "col2": [2, 3, 5, 1]}
-    modin_df3 = pd.DataFrame(frame_data, index=list("abcd"))
-
-    assert not modin_df1.equals(modin_df3)
-
-    with pytest.raises(AssertionError):
-        df_equals(modin_df3, modin_df1)
-
-    with pytest.raises(AssertionError):
-        df_equals(modin_df3, modin_df2)
+@pytest.mark.parametrize(
+    "frame1_data,frame2_data,expected_pandas_equals",
+    [
+        pytest.param({}, {}, True, id="two_empty_dataframes"),
+        pytest.param([[1]], [[0]], False, id="single_unequal_values"),
+        pytest.param([[None]], [[None]], True, id="single_none_values"),
```

Comment:
Thank you for commenting to check. The fix should work for `np.NaN` too. I added a test case for that.

Reply:
Thanks to you. Is the merge request blocked because of unsuccessful checks, or is it for other reasons?
If so, do you need help to solve the code coverage decrease and lake8 issues?

Reply:
While the solution in this PR is functionally correct (i.e. it fixes the bug), @devin-petersohn thinks it's violating some internal abstractions for Modin, so it shouldn't be merged as is. However, I don't know the proper way to make the fix. [This comment](https://github.com/modin-project/modin/pull/4148#issuecomment-1098564017) is the last one in the thread about that discussion.

If you just need a temporary fix for this issue while we get this PR merged, you could `pip install` modin from the git branch that this pull request is comparing to Modin, i.e.

`pip install git+https://github.com/mvashishtha/modin.git@4071-fix-equals-for-none`